src/constants/configSchema.js

```diff
@@ -0,0 +1,183 @@
+import AJV from 'ajv';
+import ajvErrors from 'ajv-errors';
+import {
+  formatExtensions,
+  frontmatterFormats,
+  extensionFormatters,
+} from "Formats/formats";
+import { IDENTIFIER_FIELDS } from "Constants/fieldInference";
+
+/**
+ * The schema had to be wrapped in a function to
+ * fix a circular dependency problem for WebPack,
+ * where the imports get resolved asyncronously.
+ */
+const getConfigSchema = () => ({
+  type: "object",
+  properties: {
+    backend: {
+      type: "object",
+      properties: { name: { type: "string", examples: ["test-repo"] } },
+      required: ["name"],
+    },
+    display_url: { type: "string", examples: ["https://example.com"] },
+    media_folder: { type: "string", examples: ["assets/uploads"] },
+    public_folder: { type: "string", examples: ["/uploads"] },
+    publish_mode: {
+      type: "string",
+      enum: ["editorial_workflow"],
+      examples: ["editorial_workflow"],
+    },
+    slug: {
+      type: "object",
+      properties: {
+        encoding: { type: "string", enum: ["unicode", "ascii"] },
+        clean_accents: { type: "boolean" },
+      },
+    },
+    collections: {
+      type: "array",
+      minItems: 1,
+      items: {
+        // ------- Each collection: -------
+        type: "object",
+        properties: {
+          name: { type: "string" },
+          label: { type: "string" },
+          label_singular: { type: "string" },
+          description: { type: "string" },
+          folder: { type: "string" },
+          files: {
+            type: "array",
+            items: {
+              // ------- Each file: -------
+              type: "object",
+              properties: {
+                name: { type: "string" },
+                label: { type: "string" },
+                label_singular: { type: "string" },
+                description: { type: "string" },
+                file: { type: "string" },
+                fields: {
+                  type: "array",
+                  items: {
+                    // ------- Each field: -------
+                    type: "object",
+                    properties: {
+                      name: { type: "string" },
+                      label: { type: "string" },
+                      widget: { type: "string" },
+                      required: { type: "boolean" },
+                    },
+                    required: ["name", "label"],
```

Comment:
Should we drop `label` too since that is a pending PR?

Reply:
yeah, makes sense - I'm pushing another commit right now, will include that.